src/include/common/vector/auxiliary_buffer.h

```diff
@@ -43,6 +44,14 @@ class StructAuxiliaryBuffer : public AuxiliaryBuffer {
     std::vector<std::shared_ptr<ValueVector>> childrenVectors;
 };
 
+class ArrowColumnAuxiliaryBuffer : public AuxiliaryBuffer {
+    friend class ArrowColumnVector;
+
+private:
+    common::offset_t startOffset;
```

Comment:
What is `startOffset` in a vector buffer? I don't think this is a correct design.

Reply:
As we discussed, moved startOffset to a new valueVector